File: src/map_pkg/src/map_pub_node.cpp
#include "ros/ros.h"
#include "nav_msgs/OccupancyGrid.h"

int main(int argc, char  *argv[])
{
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "map_pub_node");
    ros::NodeHandle nh;
    ros::Publisher map_pub = nh.advertise<nav_msgs::OccupancyGrid>("/map", 10);
    nav_msgs::OccupancyGrid map_msg;
    map_msg.header.frame_id = "map";
    map_msg.header.stamp = ros::Time::now();
    map_msg.info.origin.position.x = 2;
    map_msg.info.origin.position.y = 1;
    map_msg.info.resolution = 1;
    map_msg.info.width = 4;
    map_msg.info.height = 2;

    map_msg.data.resize(4*2);
    map_msg.data = {100,100,0,-1,0,0,0,0};

    ros::Rate loop_rate(1);
    while (ros::ok())
    {
        map_pub.publish(map_msg);
        loop_rate.sleep();
    }

    return 0;
}

File: src/map_pkg/scripts/map_pub_node.py
#!/usr/bin/env python3
#coding=utf8
import rospy
from nav_msgs.msg import OccupancyGrid


if __name__ == '__main__':
    rospy.init_node('map_pub_node')

    pub = rospy.Publisher("/map",OccupancyGrid,queue_size=10)
    rate = rospy.Rate(1)
    
    msg = OccupancyGrid()
    msg.header.frame_id = "map"
    msg.header.stamp = rospy.Time.now()
    msg.info.origin.position.x = 2
    msg.info.origin.position.y = 1
    msg.info.resolution = .5
    msg.info.width = 4
    msg.info.height = 2

    msg.data = [100,100,100,-1,0,0,0,0]
    while not rospy.is_shutdown():
        pub.publish(msg)
        rate.sleep()